ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -512,7 +523,18 @@ impl<'a> TraceBuilder<'a> {
         // Unwrap-safe as each frame at this point must have a safepoint associated with it.
         let mut smids = Vec::new(); // List of stackmap ids of the current call stack.
         let mut live_args = Vec::new(); // List of live JIT variables.
-        for safepoint in self.frames.iter().map(|f| (f.safepoint.unwrap())) {
+        let mut live_aot = Vec::new();
+        let mut callframes = Vec::new();
+        for frame in &self.frames {
+            let safepoint = frame.safepoint.unwrap();
+            callframes.push(Frame::new(
+                frame.callinst.clone(),
+                frame.funcidx.clone(),
+                frame.safepoint.clone(),
+                // We don't need to copy the arguments since they are only required for LoadArg
```

Comment:
We pass through here for "main traces" too don't we?

Reply:
We do. But this is only information we want to forward to side-traces and for that the arguments are irrelevant.